src/custom.rs

```diff
@@ -103,3 +103,39 @@ pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
         Some(code) => Err(Error::from(code)),
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Shouldn't these tests be defined in `tests/custom.rs` like before, since it is exercising the semantics of the public API? This way it would serve as a test that, for example, the register...! macro is defined on all targets and not just on ones for which we actually use custom implementations.

Reply:
> Shouldn't these tests be defined in tests/custom.rs.

I see now that your intent is to be able to run the tests on Linux (at least), where we don't enable the API to work, normally.

I think the best thing to do is to keep the tests/custom.rs integration test, but also have a unit test that can run on Linux. Perhaps this can be the one place we use "#[path]` tricks in the testing? Basically we'd have a "we expect custom to be ignored" test file `custom_ignored.rs` and a custom-used test file `custom_used.rs`, and in tests/* we'd select custom-ignored or custom-used based on the target OS, and in src/ we'd use `custom_used.rs` so that we can run the tests on targets for which we normally don't enable the custom feature?

Though I don't even know that that complexity is necessary because we already do have a good way to run the custom tests, via the websasembly CI runners. So I also think we could just have the custom tests only as integration tests (in `tests/` only) and test the "custom used" configuration only in webassembly for now.

Reply:
Good point. The main purposes of these tests are that a custom implementation can be registered _outside_ of this crate. So for this reason, I moved the `custom.rs` tests back to being integration tests. Given that we can run tests on unsupported platforms (via wasm CI runners), I agree that any additional complexity w.r.t. testing is unnecessary.